libraries/unit_converter.rb

```diff
@@ -0,0 +1,45 @@
+# coding: UTF-8
+#
+# Cookbook Name:: cerner_splunk
+# File Name:: unit_converter.rb
+#
+# Module contains different functions used to manipulate the units of file sizes.
+#
+module CernerSplunk
+  SIZE_SCALE = %w(KB MB GB TB).freeze
+  # Methods converts file sizes in KB, MB, GB and TB into Bytes.
+  def self.convert_to_bytes(string) # rubocop:disable CyclomaticComplexity
```

Comment:
We'd need some unit tests around it... but what about 

      REGEX = /(?i)^\s*+(\d++(?>\.\d+)?+)\s*+([kmgt](?>i?+b)?+|b?+)\s*+$/
      POWER = { '' => 0, 'B' => 0, 'K' => 1, 'M' => 2, 'G' => 3, 'T' => 4 }.freeze
    
      def self.convert_to_bytes(string)
        matchdata = string.match REGEX
        fail "Unparsable size input #{string}" unless matchdata

        size = matchdata[1].to_f
        unit = matchdata[2].upcase[0, 1]

        (size * 1024**POWER[unit]).floor
      end